book/src/Example.md

```diff
@@ -7,97 +7,227 @@ section shows how to deploy and use that driver in Kubernetes.
-by referencing the latest version directly:
-```
-$ kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/docs/master/book/src/example/csi-setup.yaml
-storageclass.storage.k8s.io "csi-hostpath-sc" created
-serviceaccount "csi-service-account" created
-clusterrole.rbac.authorization.k8s.io "csi-cluster-role" created
-clusterrolebinding.rbac.authorization.k8s.io "csi-role-binding" created
-pod "csi-pod" created
-$ kubectl get pods
-NAME      READY     STATUS    RESTARTS   AGE
-csi-pod   4/4       Running   0          24s
+This is tested with Kubernetes v1.12. Set the following feature gate flags to true:
+
+```
+--feature-dates=CSIPersistentVolume=true,MountPropagation=true,VolumeSnapshotDataSource=true,KubeletPluginsWatcher=true,CSINodeInfo=true,CSIDriverRegistry=true
+```
+
+`CSIPersistentVolume` is enabled by default in v1.10. `MountPropagation` is enabled by default
+in v1.10. `VolumeSnapshotDataSource` is a new alpha feature in v1.12. `KubeletPluginsWatcher`
+is enabled by default in v1.12. `CSINodeInfo` and `CSIDriverRegistry` are new alpha features
+in v1.12.
+
+CRD needs to be created manually for `CSIDriverRegistry` and `CSINodeInfo`.
+
+```
+$ kubectl create -f https://raw.githubusercontent.com/kubernetes/kubernetes/v1.12.0/staging/src/k8s.io/csi-api/pkg/crd/testdata/csidriver.yaml --validate=false
+customresourcedefinition.apiextensions.k8s.io/csidrivers.csi.storage.k8s.io created
+
+$ kubectl create -f https://raw.githubusercontent.com/kubernetes/kubernetes/v1.12.0/staging/src/k8s.io/csi-api/pkg/crd/testdata/csinodeinfo.yaml --validate=false
+customresourcedefinition.apiextensions.k8s.io/csinodeinfos.csi.storage.k8s.io created
+```
+
+Create RBAC rules for the CSI provisioner.
+```
+$ kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/docs/master/book/src/example/csi-provisioner-rbac.yaml
+serviceaccount/csi-provisioner created
+clusterrole.rbac.authorization.k8s.io/external-provisioner-runner created
+clusterrolebinding.rbac.authorization.k8s.io/csi-provisioner-role created
+```
+
+```yaml
+{{#include example/csi-provisioner-rbac.yaml}}
+```
+
+Create RBAC rules for the CSI attacher.
+```
+$ kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/docs/master/book/src/example/csi-attacher-rbac.yaml
+serviceaccount/csi-attacher created
+clusterrole.rbac.authorization.k8s.io/external-attacher-runner created
+clusterrolebinding.rbac.authorization.k8s.io/csi-attacher-role created
+```
+
+```yaml
+{{#include example/csi-attacher-rbac.yaml}}
+```
+
+Create RBAC rules for the node plugin.
+```
+$ kubectl create -f https://raw.githubusercontent.com/kubernetes-csi/docs/master/book/src/example/csi-nodeplugin-rbac.yaml
+serviceaccount/csi-nodeplugin created
+clusterrole.rbac.authorization.k8s.io/csi-nodeplugin created
+clusterrolebinding.rbac.authorization.k8s.io/csi-nodeplugin created
+```
+
+```yaml
+{{#include example/csi-nodeplugin-rbac.yaml}}
+```
+
+Create RBAC rules for the CSI snapshotter.
```

Comment:
Maybe point out that snapshotter is an optional piece since this is the general example documentation.

Reply:
Done